### PR TITLE
#159/#160 - Change to Palette section and the component

### PR DESCRIPTION
**Pull request by Hanna-N9**

### CODE REVIEW

**Comment by Hanna-N9** [comment]
E3755DA3-0CE0-4D05-B686-1BF007B26241_4_5005_c

File: client/src/App.js [hunk 1]
```diff
@@ -83,7 +84,9 @@ function App() {
             </button>
           </div>
         </section>
-        <section className="palette"></section>
+        <section className="palette">
+          <Palette />
```
Comment (mikelbrierly) [inline client/src/App.js:88]: Fantastic use of components to keep the code clean and readable

File: client/src/Palette.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  }, []);
+
+  return (
+    <div className="flex">
+      {paletteItems &&
+        paletteItems.map((palettes, index) => (
+          <div className="relative m-3" key={index}>
+            <div className=" flex justify-center items-center">
+              <div className="palette-circle absolute top-0 mt-10"></div>
+              <p className="absolute bottom-0 opacity-0 hover:opacity-100 bg-gray-500 w-full text-center p-2 rounded-b-2xl">
```
Comment (mikelbrierly) [inline client/src/Palette.js:21]: 👌

File: client/src/Palette.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      .then((data) => {
+        setPaletteItems(data.palettes);
+      });
+  }, []);
+
+  return (
+    <div className="flex">
+      {paletteItems &&
+        paletteItems.map((palettes, index) => (
+          <div className="relative m-3" key={index}>
```
Comment (mikelbrierly) [inline client/src/Palette.js:18]: According to the react docs, `index` should only be used as a key as a last resort. Ideally you want a unique identifier like an `ID` from the DB. Maybe we can add a TODO comment here to add a better `key` once we have a real DB connection

Comment (mikelbrierly) [reply client/src/Palette.js:18]: https://reactjs.org/docs/lists-and-keys.html#basic-list-component

**Review by mikelbrierly — COMMENTED** [review]
This is phenomenal! Just one tiny change and we can merge it in!

File: client/src/Palette.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        setPaletteItems(data.palettes);
+      });
+  }, []);
+
+  return (
+    <div className="flex">
+      {paletteItems &&
+        paletteItems.map((palettes) => (
+          // TODO: - Add key with ID after having real DB connection
+          <div className="relative m-3">
```
Comment (mikelbrierly) [inline client/src/Palette.js:19]: Sorry I think my last comment was confusing. We do want some sort of key here, but in the future, an ID will be better than the index

```suggestion
          <div className="relative m-3  key={index}">
```

**Review by mikelbrierly — CHANGES_REQUESTED** [review]
Updated a comment and added a suggestion!

**Comment by Hanna-N9** [comment]
Errors problem

**Pull request closed without merging**